debug/testlib.py

```diff
@@ -383,11 +387,16 @@ def start(self, cmd, logfile, extra_env):
             while True:
                 m = self.expect(
                         rb"(Listening on port (\d+) for gdb connections|"
-                        rb"tcl server disabled)",
+                        rb"Listening on port (\d+) for tcl connections)",
                         message="Waiting for OpenOCD to start up...")
-                if b"gdb" in m.group(1):
+                if b"gdb connections" in m.group(1):
                     self.gdb_ports.append(int(m.group(2)))
-                else:
+                elif b"tcl connections" in m.group(1):
+                    if self.tclrpc_port:
+                      raise TestLibError("unexpected re-definition of TCL-RPC port")
+                    self.tclrpc_port = int(m.group(3))
```

Comment:
I would suggest using named match groups and avoid searching a sub-string in a match:
```
                m = self.expect(
                        rb"Listening on port (?P<port>\d+) for (?P<server>(?:gdb)|(?:tcl)) connections",
                        message="Waiting for OpenOCD to start up...")
                if m["server"] == b"gdb":
                    self.gdb_ports.append(int(m["port"]))
                ...
```

Reply:
Moreover, since the regular expression has matched, `elif` can be replaced with `else` and an assertion, if you like.

Reply:
addressed. I haven't replaced `elif` with `else`, though